sotodlib/io/bookbinder.py

```diff
@@ -873,25 +1015,23 @@ def get_metadata(self, telescope=None, tube_config={}):
             tags.append(g3tobs.tag)
         meta['detsets'] = detsets
 
-        block = self.ancil.blocks['HWPEncoder_freq']
+        hwp_freq = self.ancil.hkdata.hwp_freq
         meta['hwp_freq_mean'] = None
         meta['hwp_freq_stdev'] = None
         t0, t1 = self.times[0], self.times[-1]
-        if 'approx_hwp_freq' in block.data:
-            hwp_freq = block.data['approx_hwp_freq']
-            m = (t0 < block.times) & (block.times < t1)
+        if hwp_freq is not None:
+            m = (t0 < hwp_freq.times) & (hwp_freq.times < t1)
             if m.any():
-                meta['hwp_freq_mean'] = float(np.mean(hwp_freq[m]))
-                meta['hwp_freq_stdev'] = float(np.std(hwp_freq[m]))
+                meta['hwp_freq_mean'] = float(np.mean(hwp_freq.data[m]))
+                meta['hwp_freq_stdev'] = float(np.std(hwp_freq.data[m]))
         
+        az = self.ancil.hkdata.az
         meta['az_speed_mean'] = None
         meta['az_speed_stdev'] = None
-        block = self.ancil.blocks['ACU_broadcast']
-        if 'Corrected_Azimuth' in block.data:
-            m = (t0 < block.times) & (block.times <= t1)
-            dt = np.diff(block.times[m]).mean()
-            az = block.data['Corrected_Azimuth']
-            az_speed = np.abs(np.diff(az[m]) / dt)
+        if az is not None:
+            m = (t0 < az.times) & (az.times <= t1)
+            dt = np.diff(az.times[m]).mean()
+            az_speed = np.abs(np.diff(az.data[m]) / dt)
             if m.any():
                 meta['az_speed_mean'] = float(np.mean(az_speed))
                 meta['az_speed_stdev'] = float(np.std(az_speed))
```

Comment:
My only remaining concern is how these meta are computed.  Right now there are a few possibilities:
- az is None -> speed & std are None
- az is found, but m.sum() == 0 -> speed & std are None
- az is found, but m.sum() == 1 -> speed & std are nan
- az is found, but m.sum() >= 2 -> speed & std are real numbers

Those invalid cases are non-uniform; it's annoying to have to deal with None and nan downstream, without the distinction really telling me anything definitive.  So I'd really like to see "if m.sum() >= 2" instead of "if m.any()" for all of these blocks that need to do a diff and store stats.

In practice this m.sum() == 1 won't happen very often... but then when it does, it will break a bunch of code and be annoying to retrofit.

Reply:
Ah thanks for this comment, I didn't realize that this check wasn't added. I think I correct this in the latest commit. I also made it so that the per-frame summary fields are set to `np.nan` when we can't determine them, instead of just being left out.